src/AutoUI/Filters/Filters.tsx

```diff
@@ -23,6 +27,92 @@ interface FiltersProps<T> {
 	persistFilters?: boolean;
 }
 
+const removeXNoFilterFromObj = (
+	properties: JSONSchema['properties'],
+	xNoFilter: string[] | boolean | undefined,
+): JSONSchema['properties'] => {
+	if (!properties || !Array.isArray(xNoFilter)) {
+		return properties;
+	}
+	const xNoFilterSet = new Set(xNoFilter);
+	return Object.fromEntries(
+		Object.keys(properties)
+			.filter((propKey) => !xNoFilterSet.has(propKey))
+			.map((propKey) => [propKey, properties[propKey]]),
+	);
+};
+
+const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
+	const processProperties = (
```

Comment:
I guess we don't really need to recreate this function every time, and given that we call it just once we could probably just inline it, right?